### PR TITLE
feat(dependency-dashboard): display logged repository problems 

### LINKED ISSUE #6564: Exposing Renovate logs
**What would you like Renovate to be able to do?**

We are currently setting up a bigger Renovate deployment on our internal GitLab instance. We follow a self-service model where people activate Renovate by adding it's bot user to their projects (similar to what is done for GitLab.com). Now we want to give people a way to look at Renvoate logs for repos they have access to, similar to https://app.renovatebot.com/dashboard.

Is the source code for the dashboard or some high-level docs available somewhere?

### PR DESCRIPTION
**Pull request by fgreinacher**
Closes #6564

### CODE REVIEW

**Comment by fgreinacher** [comment]
Happy to get some quick feedback whether this goes in the right direction!

**Review by viceice — COMMENTED** [review]
We need to be sure the values are sanitized, otherwise I like the idea

**Comment by rarkins** [comment]
Aren't these errors stored per-run?

Also I'm not sure we're likely to see too many of error messages per-repo. It would be very useful to store warnings.

**Comment by fgreinacher** [comment]
> Aren't these errors stored per-run?

Not sure what you mean. Can you elaborate a bit?

> Also I'm not sure we're likely to see too many of error messages per-repo. It would be very useful to store warnings.

Yes, I'll also store warnings.

**Comment by fgreinacher** [comment]
OK - this is now finally ready for a "proper" review 👓

**Comment by fgreinacher** [comment]
> We need to be sure the values are sanitized, otherwise I like the idea

They are. I added some tests to ensure that.

**Comment by fgreinacher** [comment]
> Aren't these errors stored per-run?

@rarkins I don't understand what you mean by this question, could you rephrase please?

**Comment by rarkins** [comment]
> > Aren't these errors stored per-run?
> 
> @rarkins I don't understand what you mean by this question, could you rephrase please?

I was referring to how you need to filter based on the message's `repository` instead of print all logs, but you've done that already.

File: lib/workers/repository/dependency-dashboard.ts [hunk 1]
```diff
@@ -60,6 +61,21 @@ export async function ensureMasterIssue(
...
+
+  const repoProblems = getProblems().filter(
+    (err) => err.repository === config.repository
+  );
+  if (repoProblems.length) {
+    issueBody += '## Repository problems\n\n';
+    issueBody +=
+      'The problems below occurred while renovating this repository.\n\n';
+    for (const repoProblem of repoProblems) {
+      issueBody += ` - ${nameFromLevel[repoProblem.level]}: ${
```
Comment (rarkins) [inline lib/workers/repository/dependency-dashboard.ts:73]:
```suggestion
      issueBody += ` - ${nameFromLevel[repoProblem.level].toUpperCase()}: ${
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Let's also de-dupe the printed messages. Example:

image

**Comment by fgreinacher** [comment]
> Let's also de-dupe the printed messages. Example:
> 
> image

Good point! Done with https://github.com/renovatebot/renovate/pull/6952/commits/5ac3f24c9eae614c2e7b69f6a65a2ae38833ed6b

**Comment by rarkins** [comment]
One other thing - I think artifacts errors will print at least one WARN but then it will be erased the next run. We are probably best to filter them out from the dashboard and avoid unnecessary flapping because they are preserved in the PRs anyway using comments.

**Comment by fgreinacher** [comment]
> One other thing - I think artifacts errors will print at least one WARN but then it will be erased the next run. We are probably best to filter them out from the dashboard and avoid unnecessary flapping because they are preserved in the PRs anyway using comments.

Sounds reasonable, done with https://github.com/renovatebot/renovate/pull/6952/commits/1ffe6ea7c7fbb881e366aa276c9b1b81b5587a5d.
